src/iosrv/server_iv.c

```diff
@@ -919,6 +919,8 @@ iv_op(struct ds_iv_ns *ns, struct ds_iv_key *key, d_sg_list_t *value,
 {
 	int rc;
 
+	if (ns->iv_master_rank == ((d_rank_t)-1))
+		return -DER_NOTLEADER;
```

Comment:
this seems has been checked inside ivc_on_hash(), which should be called in iv_op_internal()? I wonder if you return here, then for those retry == true case will not retry anymore in this case.

Reply:
Have we ever tested the retry logic? If this -1 master rank passed in ivc_on_hash(), it'll crash in iv code.

Reply:
seems that's a defect which has been fixed in #PR 2700 (added error checking in ds_iv_done()), so probably I can remove this check here after than patch landed.